### PR TITLE
Drop v0 contract ids

### PR DESCRIPTION
**Pull request by cocreature**
changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox-classic/src/main/scala/platform/sandbox/SandboxServer.scala [hunk 1]
```diff
@@ -381,7 +381,7 @@ final class SandboxServer(
           maxDeduplicationDurationEnforced = false,
         ),
         contractIdFeatures = ExperimentalContractIds.of(
-          v0 = ExperimentalContractIds.ContractIdV0Support.SUPPORTED,
+          v0 = ExperimentalContractIds.ContractIdV0Support.NOT_SUPPORTED,
```
Comment (cocreature) [inline ledger/sandbox-classic/src/main/scala/platform/sandbox/SandboxServer.scala:384]: I’ll drop the feature flag in a follow-up. Just trying to not increase the diff here further than necessary.

File: daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/CommandPreprocessorSpec.scala [hunk 2]
```diff
@@ -72,7 +72,7 @@ class CommandPreprocessorSpec
 
   "preprocessCommand" should {
 
-    val defaultPreprocessor = new CommandPreprocessor(compiledPackage.interface, true, false)
+    val defaultPreprocessor = new CommandPreprocessor(compiledPackage.interface, false)
```
Comment (SamirTalwar-DA) [inline daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/CommandPreprocessorSpec.scala:75]: Perhaps name the boolean parameter here, for clarity.

File: ledger/sandbox-on-x/src/test/suite/scala/com/daml/ledger/sandbox/bridge/validate/PrepareSubmissionSpec.scala [hunk 3]
```diff
@@ -48,7 +48,7 @@ class PrepareSubmissionSpec extends AsyncFlatSpec with Matchers {
     val keyValue = Value.ValueText("key-1")
 
     val createNode = txBuilder.create(
-      id = ContractId.assertFromString("#1"),
```
Comment (SamirTalwar-DA) [inline ledger/sandbox-on-x/src/test/suite/scala/com/daml/ledger/sandbox/bridge/validate/PrepareSubmissionSpec.scala:51]: Why not use the `cid` function everywhere?

**Review by SamirTalwar-DA — APPROVED** [review]

File: ledger/sandbox-on-x/src/test/suite/scala/com/daml/ledger/sandbox/bridge/validate/PrepareSubmissionSpec.scala (hunk 3 above)
Comment (SamirTalwar-DA) [reply ledger/sandbox-on-x/src/test/suite/scala/com/daml/ledger/sandbox/bridge/validate/PrepareSubmissionSpec.scala:51]: That seems fair.

**Review by remyhaemmerle-da — APPROVED** [review]
LGTM

**Pull request merged**